src/filters/detector.cpp

```diff
@@ -5,11 +5,37 @@
 namespace mesos {
 namespace serenity {
 
-Try<Nothing> DetectorFilter::consume(const ResourceUsage& in) {
+Try<Nothing> DetectorFilter::consume(const ResourceUsage& usage) {
+  this->currentVictimsUsage = usage;
+
+  if (this->currentAggressorsUsage.isSome()) {
+    this->__detect();
+  }
+
+  return Nothing();
+}
+
+
+Try<Nothing> DetectorFilter::consume(const BeResourceUsage& usage) {
```

Comment:
Could you rather consume just ResourceUsage, and then group by 'revocable' if needed?
GroupBy oneliner:
```
for (const auto & foo : foos) foos_by_x[foo.x].push_back(foo);
```
http://stackoverflow.com/questions/12335860/c-algorithm-like-pythons-groupby